internal/validation/validator.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/monitoring"
+	"github.com/canonical/identity-platform-admin-ui/internal/tracing"
+)
+
+const apiVersion = "v0"
+
+type ValidatingFunc func(r *http.Request) validator.ValidationErrors
+
+type Validator struct {
```

Comment:
Just a non-blocking question. If I understand it correctly, the `Validator` here behaves like a registry of validation functions for handlers. Do you think it is better to name it with different names (e.g. `ValidationRegistry`)? Then this will not conflict with the naming of the validator in each handler's `API` struct.

Reply:
This is a great suggestion, although with recent changes this struct will actually centralize the validation process. I think I can still use your suggestion though, I like it! Thanks